### ISSUE TITLE
Diagnostics error occurs when an @ is used in the file path

### ISSUE THREAD

**Issue by RustyMoyher** [HaxeFoundation/haxe#8086]
If the path of a file has an `@` in it, an error will occur with diagnostics.

For example, I have a folder called `@Projects`. This causes the error to occur in VSCode when I save.

`Invalid format: Projects/FlixelTut/source/PlayState.hx`

The issue appears to be a compiler bug, probably a string.split() that goes wrong. Based on discussions with @Gama11 in Discord, we believe that Haxe does a a naive string split on @, assuming that can't appear in file paths.

**Comment by Simn** [comment]
I... don't think this can be fixed in the old protocol.

**Comment by Gama11** [comment]
Why not? The `@displaymode` can only appear at the end, right?

**Comment by Simn** [comment]
Yes but the position is also separated with a `@`. It's `file@pos@mode`, _and_ the mode is optional. So with `fi@le@pos@mode` we are in a bad spot.

**Comment by Gama11** [comment]
I'm not sure that can actually happen though? Or is `@` a valid character in a module name?

The original issue here is with `@` in an earlier part of the path.

**Comment by Simn** [comment]
Not sure what you mean, something like `/something/@Projects/File.hx@12@position` is just the case I mention. Or do you think we should try to find the `File.hx` first and then take it from there?

**Comment by Gama11** [comment]
Something like that, yeah.

**Comment by kLabz** [comment]
Should be fine now that diagnostics have been moved to Json RPC
